### PR TITLE
Componentize <FilterMenu>

### LINKED ISSUE #174: Componentize <FilterMenu>
## Background

Through a series of refactoring, we'll Octanify this app and increase its maintainability.

## Problem

In the `index` and `issues` route templates, you will find duplicate code:

```handlebars
{{!-- app/templates/index.hbs, app/templates/issues.hbs --}}

<div class="help-wanted-links">
  {{#each this.navLinks.navLinks as |link|}}
    <LinkTo class="es-button" @route={{link.route}} @model={{link.model}} @query={{hash query=""}}>{{link.name}}</LinkTo>
  {{/each}}
</div>
```

## TODOs

- [ ] Use Ember CLI to create `<FilterMenu>` component. Copy-paste the following into the template:

    ```handlebars
    {{!-- app/components/filter-menu.hbs --}}

    <div class="help-wanted-links">
      {{#each this.filterNames as |filterName|}}
        <LinkTo
          class="es-button"
          data-test-link={{filterName}}
          @route="issues"
          @model={{filterName}}
          @query={{hash query=""}}
        >
          {{filterName}}
        </LinkTo>
      {{/each}}
    </div>
    ```

- [ ] In the backing class, define `filterNames`. It is an array of strings that includes the names of GitHub organizations that are related to Emberjs, plus RFCs:

    ```javascript
    // app/components/filter-menu.js

    filterNames = [
      'adopted-ember-addons',
      'ember-a11y',
      'ember-cli',
      'ember-data',
      'ember-engines',
      'ember-learn',
      'ember-template-lint',
      'emberjs',
      'empress',
      'typed-ember',
      'RFCs',
    ];
    ```

- [ ] Write a rendering test. It asserts that the component renders 11 `data-test-link`'s.

- [ ] Invoke the `<FilterMenu>` component in the `index` and `issues` route templates. Afterwards, remove the following:

    - `index` controller
    - `navLinks` injection in the `issues` controller
    - `nav-links` service

### PR DESCRIPTION
**Pull request by neojp**
This PR attempts to close #174 

- Refactor nav menu from templates `index` and `issues` into a Glimmer Component named `filter-menu`
- Refactor `nav-links` service as the `filterNames` property in the new component
- Remove `index` controller and `nav-links` service
- Add integration tests
- CSS change: Center buttons
- CSS change: Prevent menu from expanding to 100% of the viewport's width on high resolutions

### CODE REVIEW

File: app/styles/app.scss [hunk 1]
```diff
@@ -11,16 +11,11 @@ body {
 
 .help-wanted-links {
   margin-bottom: var(--spacing-4);
+  max-width: 1024px;
+  // text-align: center;
```
Comment (ijlee2) [inline app/styles/app.scss:15]: It seems like this line of code isn't needed? If so, let's remove it.

```suggestion
```

File: tests/integration/components/filter-menu-test.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import { hbs } from 'ember-cli-htmlbars';
+
+module('Integration | Component | filter-menu', function(hooks) {
+  setupRenderingTest(hooks);
+
+  test(`it renders and it's not empty`, async function(assert) {
+    await render(hbs`<FilterMenu />`);
+    assert.notEqual(this.element.textContent.trim(), '');
+  });
+
```
Comment (ijlee2) [inline tests/integration/components/filter-menu-test.js:13]: In math, we can think of the 1st test as a **weak statement**, and the 2nd test as a **strong statement**. A strong statement always implies a weak statement (assuming the statements are related).

In the context of `<FilterMenu>` component, if there is more than 1 link (what the 2nd test checks), then we know that the component is not empty (the 1st test).

We can use the idea of strong and and weak statements to remove tests and assertions that aren't necessary to write and check. Decrease time to run tests. ✨

```suggestion
```

File: tests/integration/components/filter-menu-test.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  test('buttons were rendered', async function(assert) {
+    await render(hbs`<FilterMenu />`);
+    assert.ok(findAll('[data-test-link]').length > 0);
+  });
```
Comment (ijlee2) [inline tests/integration/components/filter-menu-test.js:17]: When I check for the presence of multiple items, I like to first use `findAll` to get these items, then check for the count and test either some or all items.

```suggestion
  test('renders 11 links', async function(assert) {
    await render(hbs`
      <FilterMenu />
    `);

    const links = findAll('[data-test-link]');

    assert.strictEqual(
      links.length,
      11,
      'We see 11 links.'
    );

    assert.dom(links[0])
      .hasText(
        'adopted-ember-addons',
        'We see the correct 1st link.'
      );
  });

    assert.dom(links[links.length - 1])
      .hasText(
        'RFCs',
        'We see the correct last link.'
      );
  });
```

Again, the assertions above are stronger than the assertion `assert.ok(findAll('[data-test-link]').length > 0);` The latter assertion can pass if the component renders an incorrect number of links, whereas the suggested assertions will fail.

**Review by ijlee2 — APPROVED** [review]
Congrats on writing rendering tests for `<FilterMenu>` and `<GithubLabel>`!

They provided a good start and discussion topic. I left a couple of suggestions for how some assertions can be stronger than others and can guarantee more that the app will work as intended. Will you go over my notes some time?
